File: iOS/UI/Sources/UI/Paging/AlwaysLoadingVisibilityDecider.swift
import Foundation

public class AlwaysLoadingVisibilityDecider<Key, Item>: LoadingVisibilityDecider {
    public func shouldShowLoader(lazyPagingItems: LazyPagingItems<Key, Item>) -> Bool {
        return lazyPagingItems.loadState.refresh.isLoading
    }
}

File: Android/ui/src/main/java/com/felipearpa/tyche/ui/lazy/AlwaysLoadingVisibilityDecider.kt
package com.felipearpa.tyche.ui.lazy

import androidx.paging.LoadState
import androidx.paging.compose.LazyPagingItems

class AlwaysLoadingVisibilityDecider<Value : Any> : LoadingVisibilityDecider<Value> {
    override fun shouldShowLoader(lazyPagingItems: LazyPagingItems<Value>): Boolean =
        lazyPagingItems.loadState.refresh is LoadState.Loading
}
